generatorreceiver/internal/topology/tag_set.go

```diff
@@ -11,3 +13,25 @@ type TagSet struct {
 	Inherit             []string       `json:"inherit,omitempty" yaml:"inherit,omitempty"`
 	flags.EmbeddedFlags `json:",inline" yaml:",inline"`
 }
+
+func pickBasedOnWeight(tagSets []TagSet) TagSet {
+	totalWeight := 0.0
+	for _, ts := range tagSets {
+		if !ts.ShouldGenerate() {
+			continue // ignore disabled tagSets from calculations
+		}
+		totalWeight += ts.Weight
+	}
+	choice := rand.Float64() * totalWeight
+	current := 0.0
+	for _, ts := range tagSets {
+		if !ts.ShouldGenerate() {
```

Comment:
I don't think it'll cause any serious problems, but if you haven't already it's worth thinking through what happens here if `ts.ShouldGenerate()` changes between these two for loops.

Reply:
Nice catch!